### PR TITLE
chore: adding new command list-regions

### PR DESCRIPTION
**Pull request by gowriswarupk**
# Motiviation

Addressing issue #1386 

### Changes made:
---
- initial steps to address changes. [1]

### Verification Steps:
---

### CODE REVIEW

**Comment by wtrocki** [comment]
Marker

File: pkg/cmd/kafka/regions-new/listregions.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	"github.com/spf13/cobra"
+)
+
+type options struct {
+	id              string
+	name            string
+	bootstrapServer bool
+	outputFormat    string
+
+	IO             *iostreams.IOStreams
```
Comment (gowriswarupk) [inline pkg/cmd/kafka/regions-new/listregions.go:26]: Remove those and replace all with factory instance (f)

Comment (gowriswarupk) [reply pkg/cmd/kafka/regions-new/listregions.go:26]:
```suggestion
	f *factory.Factory
```
```

File: pkg/cmd/kafka/regions-new/listregions.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		IO:             f.IOStreams,
+		Logger:         f.Logger,
+		localizer:      f.Localizer,
+		Context:        f.Context,
+		ServiceContext: f.ServiceContext,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "list-regions",
+		Short:   opts.localizer.MustLocalize("kafka.list-regions.cmd.shortDescription"),
```
Comment (gowriswarupk) [inline pkg/cmd/kafka/regions-new/listregions.go:48]: TODO create those entries

File: pkg/cmd/kafka/regions-new/listregions.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		Long:    opts.localizer.MustLocalize("kafka.list-regions.cmd.longDescription"),
+		Example: opts.localizer.MustLocalize("kafka.list-regions.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			validOutputFormats := flagutil.ValidOutputFormats
+			if opts.outputFormat != "" && !flagutil.IsValidInput(opts.outputFormat, validOutputFormats...) {
+				return flagutil.InvalidValueError("output", opts.outputFormat, validOutputFormats...)
+			}
+
+			kafkaInstance, err := contextutil.GetCurrentKafkaInstance(f)
```
Comment (gowriswarupk) [inline pkg/cmd/kafka/regions-new/listregions.go:58]: We do not need kafka in that

File: pkg/cmd/kafka/regions-new/listregions.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+			}
+
+			opts.id = kafkaInstance.GetId()
+			return runListRegions(opts)
+		},
+	}
+
+	flags := flagutil.NewFlagSet(cmd, opts.localizer)
+
+	table := dump.TableFormat
```
Comment (gowriswarupk) [inline pkg/cmd/kafka/regions-new/listregions.go:70]: This format doesn't exist. We need to remove it.

File: pkg/cmd/kafka/regions-new/listregions.go [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+		return err
+	}
+
+	api := conn.API()
+
+	var kafkaInstance *kafkamgmtclient.KafkaRequest
+
+	providers, _, err := api.KafkaMgmt().GetCloudProviders(opts.Context).Execute()
+
+	if opts.bootstrapServer {
```
Comment (gowriswarupk) [inline pkg/cmd/kafka/regions-new/listregions.go:91]: "

**Pull request closed without merging**